### PR TITLE
string calculator

### PR DESCRIPTION
**Pull request by romos21**
Added string calculator task

### CODE REVIEW

File: string-calculator.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <link rel="stylesheet" href="assets/styles/string-calculator.css">
+    <title>Title</title>
+</head>
+<body>
+<main class="calculator">
+    <section id="window" class="calc-window"></section>
```
Comment (ghost) [inline string-calculator.html]: _UI improvements_: there is no possibility to enter the formula to some input and calculate it. For example, if I want to calculate the string "3*10" and just paste it to input.

Comment (ghost) [inline string-calculator.html]: _UI improvements_: there is no possibility to clear the input after the calculation ("C" button on a calculator)

File: assets/styles/string-calculator.css [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    cursor: pointer;
+    border-radius: 10px;
+    padding: 10px;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    background-color: #f17a1e;
+}
+
+#0,#1,#2,#3,#4,#5,#6,#7,#8,#9{
```
Comment (ghost) [inline assets/styles/string-calculator.css]: _Code style_: consider styling these elements using the class name.

File: assets/styles/string-calculator.css [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    background-color: #f17a1e;
+}
+
+#\*,#-,#\+,#\.,#\/{
```
Comment (ghost) [inline assets/styles/string-calculator.css]: _Code style_: consider styling these elements using the class name.

File: assets/js/string-calculator.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    } else if (operation === '*') {
+        return (Number(exceptionElementFirst) * Number(exceptionElementSecond)).toFixed(2);
+    } else if (operation === '/') {
+        return (Number(exceptionElementFirst) / Number(exceptionElementSecond)).toFixed(2);
+    }
+}
+
+elements.map(el => {
+    el.addEventListener('click', (event) => {
+        if(/^[-]?(\d+([.]\d+)?)[-+/*][-]?(\d+([.]\d+)?)$/g.test(windowCalc.textContent)){
```
Comment (ghost) [inline assets/js/string-calculator.js]: _Code style_: consider checking elements by class name or data-attribute

File: assets/js/string-calculator.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            if(event.target.textContent==='=') {
+                console.log(windowCalc.textContent);
+                windowCalc.textContent = calculationStart(windowCalc.textContent);
+            }else if(/[-+/*]/g.test(event.target.textContent)){
```
Comment (ghost) [inline assets/js/string-calculator.js]: _Code style_: consider checking elements by class name or data-attribute

File: assets/js/string-calculator.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+            else {
+                exceptionElementSecond+=operationCash[i];
+            }
+        }else if(!operationPointer) {
+            exceptionElementSecond+=operationCash[i];
+        }
+    }
+    console.log(exceptionElementFirst,exceptionElementSecond);
+    if (operation === '+') {
+        return Number(exceptionElementFirst) + Number(exceptionElementSecond);
```
Comment (ghost) [inline assets/js/string-calculator.js]: _Code style_: remove repetitive code (`Number(exceptionElementFirst)`, `Number(exceptionElementSecond)`)

File: assets/js/string-calculator.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+function calculationStart(exception) {
+    let operationCash = exception.split('');
+    let exceptionElementFirst='';
+    let exceptionElementSecond='';
+    let operation='';
+    exceptionElementFirst+=operationCash[0];
+    let operationPointer=true;
+    for(let i=1;i<operationCash.length;i++){
+        if(/\d+|[.]/g.test(operationCash[i]) && operationPointer){
```
Comment (ghost) [inline assets/js/string-calculator.js]: _Code style_: simplify conditional logic or move code parts to functions

File: assets/js/string-calculator.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
+const elements = [...document.getElementsByTagName('span')];
+const windowCalc = document.getElementById('window');
+
+function calculationStart(exception) {
+    let operationCash = exception.split('');
```
Comment (ghost) [inline assets/js/string-calculator.js]: _Code style_:
```
let operationCash = exception.split(''),
    exceptionElementFirst='',
    exceptionElementSecond='',
    operation='';
```

**Review by ghost — COMMENTED** [review]
@romos21 consider adding these fixes

File: assets/js/string-calculator.js (hunk 8 above)
Comment (romos21) [reply assets/js/string-calculator.js]: fixed

File: assets/js/string-calculator.js (hunk 7 above)
Comment (romos21) [reply assets/js/string-calculator.js]: fixed

File: assets/js/string-calculator.js (hunk 6 above)
Comment (romos21) [reply assets/js/string-calculator.js]: fixed

File: assets/styles/string-calculator.css (hunk 3 above)
Comment (romos21) [reply assets/styles/string-calculator.css]: fixed

File: assets/styles/string-calculator.css (hunk 2 above)
Comment (romos21) [reply assets/styles/string-calculator.css]: fixed

File: string-calculator.html (hunk 1 above)
Comment (romos21) [reply string-calculator.html]: fixed

File: assets/js/string-calculator.js (hunk 5 above)
Comment (romos21) [reply assets/js/string-calculator.js]: fixed

File: string-calculator.html (hunk 1 above)
Comment (romos21) [reply string-calculator.html]: fixed

**Pull request merged**